respx/transports.py

```diff
@@ -275,8 +275,16 @@ def record(
         response: Optional[Any],
         pattern: Optional[RequestPattern] = None,
     ) -> None:
-        request = build_request(request)
-        response = build_response(response, request=request)
+        # Decode raw request/response as HTTPX models
+        request = decode_request(request)
+        response = decode_response(response, request=request)
+
+        # TODO: Skip recording stats for pass_through requests?
+        # Pre-read response, but only if mocked, not pass-through streams streams
+        if response and not isinstance(
+            response.stream, (SyncByteStream, AsyncByteStream)
+        ):
+            response.read()
```

Comment:
I've moved the response auto reading from `decode_response`, *previously build_response*, to here since the pre-read is only for easier stats and recording.

I also needed to reverse the stream type check now that we rely on HTTPX streams and not our own `ContentStream`, which now is dropped.

@bhumilsarvaiya, could you please try this PR and verify that pass through requests still works as expected, like #68 fixed.